src/functions/sendCandidateResults/application/service/categories/AM1/custom-property-provider-cat-a-mod1.ts

```diff
@@ -0,0 +1,98 @@
+import { TestData as CatAMod1TestData } from '@dvsa/mes-test-schema/categories/AM1';
+import { CustomProperties } from '../../../../domain/custom-properties';
+import { isInteger } from 'lodash';
+import {
+  BooleanText,
+} from '../../../../domain/personalisation.model';
+
+export interface CatAMod1ICustomProperties extends CustomProperties {
+  showEmergencyStop: BooleanText;
+  showEmergencyFirstAttempt: BooleanText;
+  emergencyFirstAttempt: string;
+  showEmergencySecondAttempt: BooleanText;
+  emergencySecondAttempt: string;
+  showAvoidanceExercise: BooleanText;
+  showAvoidanceFirstAttempt: BooleanText;
+  avoidanceFirstAttempt: string;
+  showAvoidanceSecondAttempt: BooleanText;
+  avoidanceSecondAttempt: string;
+}
+
+export const getCustomPropertiesCatAMod1 = (testData: CatAMod1TestData | undefined): CatAMod1ICustomProperties => {
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  return {
+    ...getEmergencyStopAttempts(testData.emergencyStop),
+    ...getAvoidanceAttempts(testData.avoidance),
+  };
+};
+
+const getEmergencyStopAttempts = (
+  emergencyStopData: CatAMod1TestData['emergencyStop'],
+): Pick<
+  CatAMod1ICustomProperties,
+  | 'showEmergencyStop'
+  | 'showEmergencyFirstAttempt'
+  | 'emergencyFirstAttempt'
+  | 'showEmergencySecondAttempt'
+  | 'emergencySecondAttempt'
+> => {
+  const emergencyStopAttempts = {
+    showEmergencyStop: BooleanText.NO,
+    showEmergencyFirstAttempt: BooleanText.NO,
+    emergencyFirstAttempt: '',
+    showEmergencySecondAttempt: BooleanText.NO,
+    emergencySecondAttempt: '',
+  };
+
+  if (emergencyStopData) {
+    if (isInteger(emergencyStopData.firstAttempt)) {
+      emergencyStopAttempts.showEmergencyStop = BooleanText.YES;
+      emergencyStopAttempts.showEmergencyFirstAttempt = BooleanText.YES;
+      emergencyStopAttempts.emergencyFirstAttempt = emergencyStopData.firstAttempt!.toString();
+    }
+
+    if (isInteger(emergencyStopData.secondAttempt)) {
+      emergencyStopAttempts.showEmergencySecondAttempt = BooleanText.YES;
+      emergencyStopAttempts.emergencySecondAttempt = emergencyStopData.secondAttempt!.toString();
+    }
+  }
+
+  return emergencyStopAttempts;
+};
+
+const getAvoidanceAttempts = (
+  avoidanceData: CatAMod1TestData['avoidance'],
+): Pick<
```

Comment:
not seen this before - nice :)

Reply:
Yeah, this is @AndyPerryKainos . Good work!